oidc-api/src/test/java/uk/gov/di/authentication/oidc/contract/AccountInterventionServiceTest.java

```diff
@@ -0,0 +1,606 @@
+            boolean blocked, boolean suspended, boolean reproveIdentity, boolean resetPassword) {
+        return newJsonBody(
+                        (body) -> {
+                            body.object(
+                                    "intervention",
+                                    (obj) -> {
+                                        obj.numberType("updatedAt", TIME_NOW);
+                                        obj.numberType("appliedAt", TIME_NOW);
+                                        obj.numberType("sentAt", TIME_NOW);
+                                        obj.stringValue("description", "TEST_DESCRIPTION");
+                                        obj.numberType("reprovedIdentityAt", TIME_NOW);
+                                        obj.numberType("resetPasswordAt", TIME_NOW);
+                                    });
+                            body.object(
+                                    "state",
+                                    (obj) -> {
+                                        obj.booleanType("blocked", blocked);
+                                        obj.booleanType("suspended", suspended);
+                                        obj.booleanType("reproveIdentity", reproveIdentity);
+                                        obj.booleanType("resetPassword", resetPassword);
+                                    });
+                        })
+                .build();
+    }
+
+    public static class PactGenerator {
+
+        public static RequestResponsePact successfulResponse(
+                PactDslWithProvider builder,
+                boolean blocked,
+                boolean suspended,
+                boolean reproveIdentity,
+                boolean resetPassword) {
+            return builder.given("internal-pairwise-subject-id is valid")
+                    .uponReceiving(
+                            String.format(
+                                    "A request for an account that has state: blocked = %s, suspended = %s, reproveIdentity = %s, resetPassword = %s",
+                                    blocked, suspended, reproveIdentity, resetPassword))
+                    .path("/v1/ais/" + INTERNAL_PAIRWISE_SUBJECT_ID)
+                    .method("GET")
+                    .willRespondWith()
+                    .status(200)
+                    .body(createAisResponseBody(blocked, suspended, reproveIdentity, resetPassword))
+                    .toPact();
+        }
+
+        public static RequestResponsePact errorResponse(PactDslWithProvider builder, int status) {
+            return builder.uponReceiving(
+                            "A request that results in an error response with status " + status)
+                    .path("/v1/ais/" + INTERNAL_PAIRWISE_SUBJECT_ID)
+                    .method("GET")
+                    .willRespondWith()
+                    .status(status)
+                    .body("")
+                    .toPact();
+        }
+    }
```

Comment:
We should definitely be unit testing these error cases, but I'm not sure how much value there is in Pact testing all of them. The contract we're forming is that we both do the correct things based on the scenario we're setting up. It's going to be really hard to write a provider test for a 502, and not very valuable. 

On the other hand, a contact that says something like "given 1234 does not appear in your database, upon receiving a request for 1234 we receive a response of false,false,false,false"